### PR TITLE
Finesse workout input

### PR DESCRIPTION
**Pull request by thomasleese**
- Makes the edit/delete buttons smaller on phones.
- Improves the title of workouts by removing the date year.
- Adds links to the exercises where possible.

### CODE REVIEW

File: workout/models.py [hunk 1]
```diff
@@ -52,20 +52,8 @@ def title(self):
 
     @property
     def duration(self):
-        try:
```
Comment (vikashtank) [inline workout/models.py]: did this remove the duration property?

File: workout/models.py [hunk 2]
```diff
@@ -38,14 +38,17 @@ def number_of_exercises(self):
...
 
-        muscles_worked = ', '.join(group_names)
-        date = self.start_time.strftime('%d %b %Y')
-
-        return f'{date} - {muscles_worked}'
+        if group_names:
+            muscles_worked = ', '.join(group_names)
+            return f'{muscles_worked} - {date}'
+        else:
+            return date
```
[the remark this answers is not in the document]
Comment (thomasleese) [reply workout/models.py]: I can refactor that... if you insist!

File: workout/templates/includes/workout/workout_card.html [hunk 3]
```diff
@@ -18,8 +18,25 @@ <h5 class="card-title mb-0">
     </div>
   </div>
 
+  {% load workout_extras %}
```
Comment (vikashtank) [inline workout/templates/includes/workout/workout_card.html]: what is this load?

File: workout/models.py (hunk 1 above)
Comment (thomasleese) [reply workout/models.py]: Nah, I changed duration to return the time now - start time if the workout wasn’t finished, the property is still there.

File: workout/templatetags/workout_extras.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    total_seconds = duration.total_seconds()
+    hours, remainder = divmod(total_seconds, 60 * 60)
+    minutes, seconds = divmod(remainder, 60)
+
+    minutes_seconds_str = f'{int(minutes)} minutes {int(seconds)} seconds'
+
+    if hours == 0:
+        return minutes_seconds_str
+    else:
+        return f'{int(hours)} hours {minutes_seconds_str}'
```
Comment (vikashtank) [inline workout/templatetags/workout_extras.py]: cough multiple returns cough

File: workout/templates/includes/workout/workout_card.html (hunk 3 above)
Comment (thomasleese) [reply workout/templates/includes/workout/workout_card.html]: It’s a new file in the templatetags folder, I’ve moved the display_duration properly to make it a template tag, so we can do:

{{ thingy.duration | duration }}

It needs this load to get access to the duration thing after the |

Comment (vikashtank) [reply workout/templates/includes/workout/workout_card.html]: did you make it into template tags because it belongs to the views?

Comment (thomasleese) [reply workout/templates/includes/workout/workout_card.html]: Yep

File: workout/templates/includes/workout/buttons/delete_set.html [hunk 5]
```diff
@@ -1,6 +1,7 @@
-<a class="btn btn-danger btn-sm" href="{% url 'workout:delete_set' pk=set.pk %}" data-toggle="modal" data-target="#deleteSetModal{{ set.id }}">
-  <span class="fa fa-times mr-1"></span> Delete
+{% url 'workout:delete_set' pk=set.pk as url_path %}
+
+<a class="btn btn-outline-danger btn-sm" href="{{ url_path }}" data-toggle="modal" data-target="#deleteSetModal{{ set.id }}">
+  <span class="fa fa-trash"></span> <span class="d-none d-sm-none d-md-inline ml-md-1">Delete</span>
 </a>
```
Comment (vikashtank) [inline workout/templates/includes/workout/buttons/delete_set.html]: think delete_set and delete_exercise can be joined?

File: workout/templates/includes/workout/workout_card.html (hunk 3 above)
Comment (vikashtank) [reply workout/templates/includes/workout/workout_card.html]: thumbs up*

Comment (vikashtank) [reply workout/templates/includes/workout/workout_card.html]: I approve violently

**Pull request merged**